File: arrays and hashing/Two Sum.py
def twoSum(nums: list[int], target: int) -> list[int]:
    diffs = {}

    for i, num in enumerate(nums):
        diff = target - num
        if num in diffs:
            return [diffs[num], i]
        else:
            diffs[diff] = i

    return []

print(twoSum([3,4,5,6], 7))

File: arrays and hashing/Two Sum.cs
public class SolutionTwoSum
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> sumNeeded = new Dictionary<int, int>(); // Key: remainder we are looking for  Value: index unique
        for (int i = 0; i < nums.Length; i++)
        {
            if (sumNeeded.ContainsKey(target - nums[i]))
                return new int[2] { i, sumNeeded[target - nums[i]] };
            sumNeeded[nums[i]] = i;
        }

        return new int[2] { -1, -1 };
    }
}
